### ISSUE TITLE
' easy of use' features for content viewed online

### ISSUE THREAD

**Issue by steveowashere** [haiwen/seahub#1819]
Hello! Just wondering if it's possible to add some simple 'easy of use' features to the viewers for online? Just simple tweaks that would make using Seahub a bit more convenient.  

- Ability to zoom and rotate PDFs while viewing them. [Mockup image](https://i.imgur.com/xZ6RFvG.png)

- Ability to rotate PDFs.[Mockup image](https://i.imgur.com/oB6bfdJ.png)

- Page counter for PDF view, with ability to type in page number

- Select several music files and play them in a playlist of sorts. [Mockup image](https://i.imgur.com/oKsa34H.png). (Alternatively, when several music files are selected, clicking on one of them will play all)

- Spell check for the markdown editor. (I understand this might be really tough to do, because spell checking in javascript = evil?)

- Some type of image gallery that can view a whole folder of images in full screen. 

Thanks for all the hardwork on Seafile/Seahub, it's by far the best cloud storage i've used.

**Comment by shoeper** [comment]
> Spell check for the markdown editor.

Your browser likely can do that for you.

**Comment by steveowashere** [comment]
> Your browser likely can do that for you.

It can't sadly. It's actually a problem with a lot of Markdown editors I've used in browser. Tried multiple browsers and Windows and Linux of course.

**Comment by shoeper** [comment]
Looks like it is possible with html5 to enable spellchecking of the browser using `contenteditable="true" spellcheck="true` and supported by all browsers. Would require an enhancement of editor.md, though.

**Comment by steveowashere** [comment]
Seems 'simple' enough then. Would be a really nice tweak to have.

**Comment by castilma** [comment]
> Some type of image gallery that can view a whole folder of images in full screen.

yes please! on a desktop screen the preview is large enough. but on mobile...
related to #356?

**Comment by steveowashere** [comment]
The mobile app actually can sorta do this. If you click on a picture it will let you see it full screen and you can swipe back and forth between pictures. (At least on Android, never tried the iOS version of seafile.)
